go/understack/cmd/root.go

```diff
@@ -20,10 +34,66 @@ var RootCmd = &cobra.Command{
 	},
 }
 
+func expandPath(path string) (string, error) {
+	if strings.HasPrefix(path, "~") {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			return "", err
+		}
+		return filepath.Join(home, path[1:]), nil
+	}
+	return path, nil
+}
+
+func preRun(cmd *cobra.Command, args []string) error {
+	deployRepo := viper.GetString(deployRepoFlag)
+	if deployRepo == "" {
+		return nil
+	}
+
+	expandedRepoDir, err := expandPath(deployRepo)
+	if err != nil {
+		return fmt.Errorf("failed to expand path: %w", err)
+	}
+
+	info, err := os.Stat(expandedRepoDir)
+	if err != nil {
+		return fmt.Errorf("could not access directory '%s': %w", expandedRepoDir, err)
+	}
+	if !info.IsDir() {
+		return fmt.Errorf("path '%s' is not a directory", expandedRepoDir)
+	}
+
+	// Switch working directory
+	if err := os.Chdir(expandedRepoDir); err != nil {
+		return fmt.Errorf("failed to change working directory to '%s': %w", expandedRepoDir, err)
+	}
+
+	log.Printf("deployment repo path: %s", expandedRepoDir)
+
+	return nil
+}
+
 // Execute will execute the root command
 func Execute() error {
 	return RootCmd.Execute()
 }
 
 func init() {
+	// bind our flag
+	if err := viper.BindEnv(deployRepoFlag, deployRepoEnvVar); err != nil {
+		panic(fmt.Sprintf("failed to bind %s env var: %v", deployRepoFlag, err))
```

Comment:
Instead of panic we can simply do.
```
log.Fatal("failed to bind ", "env", deployRepoFlag, "err", err)
os.Exit(1)
```
output will be like this.

![image](https://github.com/user-attachments/assets/d278a290-f1fb-4774-970b-42bd2ff0a17b)